### PR TITLE
Add --sync-grains CLI option

### PR DESCRIPTION
**Pull request by mirceaulinic**

### CODE REVIEW

File: salt_sproxy/cli.py [hunk 1]
```diff
@@ -195,15 +204,17 @@ def run(self):
...
         }
         for opt, kwarg in six.iteritems(reverse_opts):
             if getattr(self.options, opt):
                 kwargs[kwarg] = False
         kwargs['events'] = self.config.get('events', False)
         kwargs['use_existing_proxy'] = self.config.get('use_existing_proxy', False)
         kwargs['test_ping'] = self.config.get('test_ping', False)
-        kwargs['target_cache_timeout'] = self.config.get('target_cache_timeout', 60)  # seconds
+        kwargs['target_cache_timeout'] = self.config.get(
+            'target_cache_timeout', 60
```
Comment (isbm) [inline salt_sproxy/cli.py]: Aren't these defaults should be beforehand prepared in `self.config`?

File: salt_sproxy/parsers.py [hunk 2]
```diff
@@ -239,13 +239,21 @@ def _mixin_setup(self):
             help='Show the devices expected to match the target.',
         )
         self.add_option(
-            '--sync-modules',
-            dest='sync_modules',
+            '--sync-grains',
+            dest='sync_grains',
```
Comment (isbm) [inline salt_sproxy/parsers.py]: I am wondering if `--sync-modules --sync-grains --sync-pillars` etc is better than `--sync m,g,p` or `--sync mods,grains,pillars` whatsoever.

File: salt_sproxy/parsers.py [hunk 3]
```diff
@@ -323,9 +331,7 @@ def _mixin_setup(self):
             dest='no_target_cache',
             action='store_true',
             default=False,
-            help=(
-                'Do no load the list of targets from the cache.'
-            ),
+            help=('Do no load the list of targets from the cache.'),
```
Comment (isbm) [inline salt_sproxy/parsers.py]: `Do not`.

File: salt_sproxy/_modules/netbox.py [hunk 4]
```diff
@@ -125,6 +123,4 @@ def get(app, endpoint, id=None, **kwargs):
         return dict(getattr(getattr(nb, app), endpoint).get(id))
     else:
         clean_kwargs = __utils__['args.clean_kwargs'](**kwargs)
-        return dict(
-            getattr(getattr(nb, app), endpoint).get(**clean_kwargs)
-        )
+        return dict(getattr(getattr(nb, app), endpoint).get(**clean_kwargs))
```
Comment (isbm) [inline salt_sproxy/_modules/netbox.py]: This is pretty scary as it is very fragile and might be a great source of strange bugs. I would rather break that to defaults and fail, if a method was not found, something like:

``python
ret = None
app = getattr(nb, app, None)
if app is not None:
    endpoint = getattr(app, endpoint, None)
    if endpoint is not None:
        ret = dict(endpoint.get(**clean_kwargs))
    else:
        log.error("Suspicious pointer corrupted virtual machine")
else:
    log.error("The recent proliferation of Nuclear Testing")
return ret

File: salt_sproxy/cli.py (hunk 1 above)
Comment (mirceaulinic) [reply salt_sproxy/cli.py]: Nah, in `self.config` it's only what's in the config file + CLI options.

File: salt_sproxy/parsers.py (hunk 2 above)
Comment (mirceaulinic) [reply salt_sproxy/parsers.py]: That might be a good idea. I think I incline to add these one individually, i.e., `--sync-grains` etc. together with that alternative you suggested.

File: salt_sproxy/_modules/netbox.py (hunk 4 above)
Comment (mirceaulinic) [reply salt_sproxy/_modules/netbox.py]: I hear ya, I agree. However, this module is simply ported from https://github.com/saltstack/salt/blob/develop/salt/modules/netbox.py with some minor corrections (that are in some PRs that didn't get too much love). Ideally, this module shouldn't be here at all, but it's kinda required and I thought I'd port it with a minimum of changes from the "official" one.

File: salt_sproxy/parsers.py (hunk 2 above)
Comment (isbm) [reply salt_sproxy/parsers.py]: That's just an idea, surely not for _this_ PR. :wink:

**Pull request merged**